### ISSUE TITLE
Expose match count/index for modeline segments?

### ISSUE THREAD

**Issue by indigoviolet** [raxod502/ctrlf#120]
Does ctrlf expose its current search state anywhere for use in segments like https://github.com/seagle0128/doom-modeline/blob/master/doom-modeline-segments.el#L1222 ? 

My experimentation and skimming of the code suggested that it constructs its overlays lazily, so that I can't just count the number of overlays.

**Comment by raxod502** [comment]
I do not believe these stats are exposed in a friendly way, but they could/should be. Sorry for the inconvenience.
